modules/service/src/main/scala/lucuma/odb/service/GmosSequenceService.scala

```diff
@@ -26,6 +27,14 @@ import Services.Syntax.*
 
 trait GmosSequenceService[F[_]] {
 
+  def selectGmosNorthStepRecords(
+    observationId: Observation.Id
+  ): Stream[F, StepRecord[DynamicConfig.GmosNorth]]
```

Comment:
@tpolecat Dumb question: does the creation of the `Stream` of steps need to be done with a `Transaction`, or is it only needed when the `Stream` is actually executed?  In other words, should this be `using Transaction`?

Reply:
It's only needed on execution; the transaction resource, when used, conceptually just puts `begin` and `commit` around whatever you're doing. So as long as you compose this with a transaction eventually you'll be fine.